cura/CuraApplication.py

```diff
@@ -1241,3 +1242,23 @@ def _readMeshFinished(self, job):
 
     def addNonSliceableExtension(self, extension):
         self._non_sliceable_extensions.append(extension)
+
+    @pyqtSlot(str, result = bool)
+    def checkIsProjectFile(self, file_name):
+        """
+        Checks if the given file is a valid project file.
+        """
+        is_valid_project_file = False
+        if file_name.startswith('file:///'):
+            file_name = file_name[8:]
+        try:
+            with zipfile.ZipFile(file_name) as zf:
```

Comment:
Is this the best way to determine if the file is a project file? It may be sufficient to find a "Cura" folder?